### PR TITLE
Add PayloadInterface.

### PR DESCRIPTION
**Pull request by dolfelt**
@elazar @shadowhand Initial definition of `PayloadInterface`.

### CODE REVIEW

File: src/PayloadInterface.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * Create a new payload with the status.
+     *
+     * @param int $status
+     * @return PayloadInterface
+     */
+    public function withStatus($status);
+
+    /**
+     * Get the status from the payload.
```
Comment (shadowhand) [inline src/PayloadInterface.php]: I would say `of the` instead of `from the`. Just nitpicking.

File: src/PayloadInterface.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Spark\Adr;
+
+interface PayloadInterface
+{
+    /**
+     * Create a new payload with the status.
```
Comment (shadowhand) [inline src/PayloadInterface.php]: I would say:

> Create a copy of the payload with the given status.

File: src/PayloadInterface.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Get the status from the payload.
+     *
+     * @return int
+     */
+    public function getStatus();
+
+    /**
+     * Create a new payload with output array.
```
Comment (shadowhand) [inline src/PayloadInterface.php]: Similar here and elsewhere, including "copy" seems important to me.

**Comment by shadowhand** [comment]
Being pedantic about language aside, interface looks correct in functionality.

**Comment by dolfelt** [comment]
Updated :abc:. :rocket:

File: src/PayloadInterface.php (hunk 1 above)
Comment (dolfelt) [inline src/PayloadInterface.php]: I love nitpicking! :smile:

File: src/PayloadInterface.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * Create a copy of the payload with output array.
+     *
+     * @param array $output
+     * @return PayloadInterface
+     */
+    public function withOutput(array $output);
+
+    /**
+     * Get output array from the payload
```
Comment (shadowhand) [inline src/PayloadInterface.php]: Missing a period on this line.

File: src/PayloadInterface.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * Create a copy of the payload with messages array.
+     *
+     * @param array $output
+     * @return PayloadInterface
+     */
+    public function withMessages(array $messages);
+
+    /**
+     * Get messages array from the payload
```
Comment (shadowhand) [inline src/PayloadInterface.php]: Missing period here too.

**Comment by shadowhand** [comment]
Couple more nitpicks, then... :shipit:

**Pull request merged**